### PR TITLE
feature/AKD-118: Added Azure 'DDoS Standard Protection Enabled' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'DDoS Standard Protection Enabled',
+    category: 'Virtual Networks',
+    description: 'Ensures that DDoS Standard Protection is enabled for Microsoft Azure Virtual Networks',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js:7]:
```suggestion
    description: 'Ensure that DDoS Standard Protection is enabled for Microsoft Azure Virtual Networks.',
```

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    recommended_action: 'Enable DDoS protection for virtual networks',
+    link: 'https://azure.microsoft.com/en-us/blog/azure-ddos-protection-for-virtual-networks-generally-available/',
+    apis: ['virtualNetworks:listAll'],
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.servers, (location, rcb) => {
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js:18]:
```suggestion
        async.each(locations.virtualNetworks, (location, rcb) => {
```

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            listAll: {
+                'eastus': {}
+            }
+        }
+    };
+};
+
+describe('ddosStandardProtectionEnabled', function() {
+    describe('run', function() {
+        it('should give passing result if no servers', function(done) {
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js:49]:
```suggestion
        it('should give passing result if no virtual networks', function(done) {
```

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            ddosStandardProtectionEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('DDoS Standard Protection is not enabled for Microsoft Azure Virtual Network');
+                expect(results[0].region).to.equal('eastus');
+                done();
+            });
+        });
+
+        it('should give unknown result if DDoS standard protection is not enabled for virtual network', function(done) {
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js:71]:
```suggestion
        it('should give unknown result if unable to query for virtual networks', function(done) {
```

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        name: 'test-vnet',
+        id: '/subscriptions/123/resourceGroups/aqua-resource-group/providers/Microsoft.Network/virtualNetworks/test-vnet',
+        type: 'Microsoft.Network/virtualNetworks',
+        location: 'eastus',
+        provisioningState: 'Succeeded',
+        virtualNetworkPeerings: [],
+        enableDdosProtection: false
+    }
+];
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js:23]: Use double quotes notation to define objects
#586

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js (hunk 1 above)
Comment (giorod3) [reply plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js:7]: I like Ensures better.

**Review by giorod3 — COMMENTED** [review]
after the changes @AkhtarAmir requested this looks good to go

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js (hunk 2 above)
Comment (ali-imran7) [reply plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js]: Done as suggested.

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js (hunk 3 above)
Comment (ali-imran7) [reply plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js]: Done as suggested.

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js (hunk 4 above)
Comment (ali-imran7) [reply plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js]: Done as suggested.

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js (hunk 5 above)
Comment (ali-imran7) [reply plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.spec.js]: Done as suggested.

File: plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+            if (!virtualNetworks) return rcb();
+
+            if (virtualNetworks.err || !virtualNetworks.data) {
+                helpers.addResult(results, 3, 'Unable to query for Virtual Networks: ' + helpers.addError(virtualNetworks), location);
+                return rcb();
+            }
+
+            if (!virtualNetworks.data.length) {
+                helpers.addResult(results, 0, 'No existing Virtual Networks found', location);
```
Comment (giorod3) [inline plugins/azure/virtualnetworks/ddosStandardProtectionEnabled.js:30]:
```suggestion
                helpers.addResult(results, 0, 'No existing Virtual Networks found', location);
                return rcb();
```

**Pull request merged**